tree/dataframe/test/dataframe_vary.cxx

```diff
@@ -125,6 +136,33 @@ TEST(RDFVary, VariationsForDoesNotTriggerRun)
    EXPECT_EQ(df.GetNRuns(), 0);
 }
 
+TEST(RDFVary, InvalidQueries)
+{
+   auto df = ROOT::RDataFrame(10).Define("x", [] { return 1.f; });
+   // x is float, variation expression cannot return RVec<int>, must be RVec<float>
+   EXPECT_THROW(df.Vary("x", SimpleVariation, {}, 2), std::runtime_error);
```

Comment:
Here, and on some other places I *think* it is a good idea to check what is the exact message thrown. I did not do that previously, as we were continuously changing few of the error messages.

I am taking care of it now.